src/main/java/com/example/delivery_system/service/CarriageService.java

```diff
@@ -0,0 +1,55 @@
+package com.example.delivery_system.service;
+
+import com.example.delivery_system.entity.Carriage;
+import com.example.delivery_system.repository.CarriageRepository;
+import com.example.delivery_system.repository.OrderRepository;
+import com.example.delivery_system.repository.RouteRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class CarriageService {
+
+    private final CarriageRepository carriageRepository;
+    private final RouteRepository routeRepository;
+    private final OrderRepository orderRepository;
+
+    public Carriage findCarriageById(Long carriageId) {
+        Optional<Carriage> carriageFromDb = carriageRepository.findById(carriageId);
+        return carriageFromDb.orElse(new Carriage());
```

Comment:
If you do not find a record you need to throw an exception and then handle it as a 404 status code. I would prefer to do it with @ControllerAdvice https://www.baeldung.com/exception-handling-for-rest-with-spring#controlleradvice